### PR TITLE
feat(refs DPLAN-12697): add custom html exchange format for prosemirror data

### PR DESCRIPTION
**Pull request by muellerdemos**
### Ticket
https://demoseurope.youtrack.cloud/issue/DPLAN-12697/Abstimmung-Austauschformat-Slicing-Tagging-PRE-AI

**WIP**
#### what is currently working:
- tests :)
- basic transformation of prosemirror formatted data into a html string and vice versa
#### what is missing:
- make this clean and remove redundancies
- not all parsed attributes are available currently (e.g. segment_text)

Basically a PoC state but certain aspects already functional even though not cleaned up yet. But still quite a bit to do:
- make sure the prosemirror data is valid and useable in editors
- make sure the html string contains all relevant data

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly
- [ ] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

File: client/js/lib/prosemirror/converter.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * @param htmlString - string
+ */
+export class ProseMirrorConverter {
+  constructor() {
+    this.prosemirrorData = null
+    this.htmlString = null
+  }
+
+  /**
+   * Returns a ProseMirrorConverter instance filled with ProseMirror data.
```
Comment (spiess-demos) [inline client/js/lib/prosemirror/converter.js:20]: Nitpick: At this point it is not perfectly clear what "ProseMirror data" may be.

File: client/js/lib/prosemirror/converter.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/**
+ * This ProseMirrorConverter class is responsible for converting
+ * ProseMirror data to an HTML string and to convert proseMirror data
+ * to an HTML string.
```
Comment (spiess-demos) [inline client/js/lib/prosemirror/converter.js:4]: Nitpick: something seems to be duplicated here.

File: client/js/lib/prosemirror/converter.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  /**
+   * Returns a ProseMirrorConverter instance filled with an HTML string.
+   * @param htmlString
+   * @returns { ProseMirrorConverter }
+   */
+  fromHtml(htmlString) {
+    try {
+      const parser = new DOMParser()
```
Comment (spiess-demos) [inline client/js/lib/prosemirror/converter.js:55]: I'd suggest to instanciate the parser with the class constructor so it is not instanciated everytime again, when `fromHtml` runs.

File: client/js/lib/prosemirror/converter.js (hunk 1 above)
Comment (muellerdemos) [reply client/js/lib/prosemirror/converter.js]: I see, I tried to be more explicit here: https://github.com/demos-europe/demosplan-core/pull/3875/commits/4ba7fe8b305eb22aa506d0e9ae95102779fa1faa and hope by adding a documentation link too, things will be more clear.

File: client/js/lib/prosemirror/converter.js (hunk 2 above)
Comment (muellerdemos) [reply client/js/lib/prosemirror/converter.js]: Ups, thanks a lot, this was adjusted here, too https://github.com/demos-europe/demosplan-core/pull/3875/commits/4ba7fe8b305eb22aa506d0e9ae95102779fa1faa

File: client/js/lib/prosemirror/converter.js (hunk 3 above)
Comment (muellerdemos) [reply client/js/lib/prosemirror/converter.js]: This is a really great suggestion, thanks a lot!
Done here: https://github.com/demos-europe/demosplan-core/pull/3875/commits/269959511ba697df98843ec0a812649697c6cd31

**Comment by muellerdemos** [comment]
@spiess-demos thank you for having an early look at this piece of work already. I was also not so sure if we want this class syntax since I rarely see this done in dplan projects (I personally like it from time to time since it enables method chaining quite easily) but I am not sure if this is a shared view.

File: client/js/lib/prosemirror/converter.js (hunk 1 above)
Comment (spiess-demos) [reply client/js/lib/prosemirror/converter.js]: thanks! i think this makes it clear for "first time readers" of that code.

**Comment by muellerdemos** [comment]
Unfortunately this piece of work was redundant hence this PR can be deleted. We never need to transform the exchange format from html to prosemirror or vice versa. The format will be provided by data and we need to use it then.

**Pull request closed without merging**